pkg/cmd/server/server_test.go

```diff
@@ -69,3 +70,80 @@ func TestVeleroResourcesExist(t *testing.T) {
 	veleroAPIResourceList.APIResources = veleroAPIResourceList.APIResources[:3]
 	assert.Error(t, server.veleroResourcesExist())
 }
+
+func TestRemoveControllers(t *testing.T) {
+	logger := velerotest.NewLogger()
+
+	tests := []struct {
+		name                string
+		disabledControllers []string
+		errorExpected       bool
+	}{
+		{
+			name: "Remove one disabable controller",
+			disabledControllers: []string{
+				controller.Backup,
+			},
+			errorExpected: false,
+		},
+		{
+			name: "Remove all disabable controllers",
+			disabledControllers: []string{
+				controller.Backup,
+				controller.BackupDeletion,
+				controller.BackupSync,
+				controller.DownloadRequest,
+				controller.GarbageCollection,
+				controller.ResticRepo,
+				controller.Restore,
+				controller.Schedule,
+				controller.ServerStatusRequest,
+			},
+			errorExpected: false,
+		},
+		{
+			name: "Remove with a non-disabable controller included",
+			disabledControllers: []string{
+				controller.Backup,
+				controller.BackupStorageLocation,
+			},
+			errorExpected: true,
+		},
+		{
+			name: "Remove with a misspelled controller name",
+			disabledControllers: []string{
+				"bakup",
+			},
+			errorExpected: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			enabledControllers := map[string]func() controllerRunInfo{
+				controller.BackupSync:        func() controllerRunInfo { return controllerRunInfo{} },
+				controller.Backup:            func() controllerRunInfo { return controllerRunInfo{} },
+				controller.Schedule:          func() controllerRunInfo { return controllerRunInfo{} },
+				controller.GarbageCollection: func() controllerRunInfo { return controllerRunInfo{} },
+				controller.BackupDeletion:    func() controllerRunInfo { return controllerRunInfo{} },
+				controller.Restore:           func() controllerRunInfo { return controllerRunInfo{} },
+				controller.ResticRepo:        func() controllerRunInfo { return controllerRunInfo{} },
+				controller.DownloadRequest:   func() controllerRunInfo { return controllerRunInfo{} },
+			}
+
+			enabledRuntimeControllers := map[string]struct{}{
+				controller.ServerStatusRequest: {},
+			}
+
+			totalNumOriginalControllers := len(enabledControllers) + len(enabledRuntimeControllers)
+
+			if tt.errorExpected {
+				assert.Error(t, removeControllers(tt.disabledControllers, enabledControllers, enabledRuntimeControllers, logger))
+			} else {
+				assert.NoError(t, removeControllers(tt.disabledControllers, enabledControllers, enabledRuntimeControllers, logger))
+
+				totalNumEnabledControllers := len(enabledControllers) + len(enabledRuntimeControllers)
```

Comment:
Optional, but I think it would be nice if instead of checking the counts, we iterate over the `disabledControllers` slice and check that that key doesn't appear in the `enabledControllers` or `enabledRuntimeControllers` map.

Reply:
I agree. I left the testing of the count too. ✅